### PR TITLE
Use Cake Router to generate JSON API pagination links

### PR DESCRIPTION
**Pull request by bravo-kernel**
This is the last PR on my list and fixes the pagination links by generating them using CakePHP Router instead of lame hackery (was forgotten in #467).

### CODE REVIEW

**Comment by jippi** [comment]
lgtm

File: src/Listener/ApiPaginationListener.php [hunk 1]
```diff
@@ -87,27 +87,33 @@ public function beforeRender(Event $event)
      */
     protected function _getJsonApiPaginationResponse(array $pagination)
     {
-        $self = Router::url([
+        if ($this->_controller()->Crud->config('listeners.jsonApi.absoluteLinks') === true) {
+            $routerMethod = 'url';
+        } else {
```
Comment (davidyell) [inline src/Listener/ApiPaginationListener.php]: The variables default state could be assigned regardless, and only if the condition matches would it be overwritten, which would remove the `else`. Just a thought 😄

**Comment by bravo-kernel** [comment]
That's what I had originally, reverting 😏

Sent from my Mobile Device

**Comment by bravo-kernel** [comment]
Changed as suggested, rebased, feel free to merge ;)

**Comment by bravo-kernel** [comment]
Thanks!

**Pull request merged**